buildpipeline/perf-pipeline.groovy

```diff
@@ -6,9 +6,395 @@
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x64.Windows_NT x64ThroughputBenchmarks"
+        bat "move Microsoft.BenchView.ThroughputBenchmarks.x86.Windows_NT x86ThroughputBenchmarks"
+
+        stash includes: 'Microsoft.BenchView.JSONFormat/**/*', name: 'benchview-tools'
+        stash name: "metadata", includes: "*.json"
+        stash name: "throughput-benchmarks-x64", includes: "x64ThroughputBenchmarks/**/*"
+        stash name: "throughput-benchmarks-x86", includes: "x86ThroughputBenchmarks/**/*"
+    }
+}
+
+// TODO: use non-pgo builds for throughput?
+def innerLoopBuilds = [
+    "windows x64 pgo build": {
+        simpleNode('Windows_NT','latest') {
+            windowsBuild('x64', config, 'pgo', false)
+        }
+     },
+    "windows x86 pgo build": {
+        simpleNode('Windows_NT','latest') {
+            windowsBuild('x86', config, 'pgo', false)
+        }
+    },
+    "linux x64 pgo build": {
+        simpleNode('RHEL7.2', 'latest-or-auto') {
+            linuxBuild('x64', config, 'pgo', false)
+        }
     }
+]
+
+// Only run non-pgo builds on offical builds
+def outerLoopBuilds = [:]
+
+if (!isPR()) {
+    outerLoopBuilds = [
+        "windows x64 nopgo build": {
+            simpleNode('Windows_NT','latest') {
+                windowsBuild('x64', config, 'nopgo', false)
+            }
+        },
+        "windows x86 nopgo build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x86', config, 'nopgo', false)
+           }
+        },
+        "linux x64 nopgo build": {
+           simpleNode('RHEL7.2', 'latest-or-auto') {
+               linuxBuild('x64', config, 'nopgo', false)
+           }
+        }
+    ]
+}
+
+def baselineBuilds = [:]
+
+if (isPR()) {
+   baselineBuilds = [
+       "windows x64 pgo baseline build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x64', config, 'pgo', true)
+           }
+       },
+       "windows x86 pgo baseline build": {
+           simpleNode('Windows_NT','latest') {
+               windowsBuild('x86', config, 'pgo', true)
+           }
+       },
+       "linux x64 pgo baseline build": {
+           simpleNode('RHEL7.2', 'latest-or-auto') {
+               linuxBuild('x64', config, 'pgo', true)
+           }
+       }
+   ]
+}
+
+stage ('Build Product') {
+    parallel innerLoopBuilds + outerLoopBuilds + baselineBuilds
 }
 
+// Pipeline builds don't allow outside scripts (ie ArrayList.Add) if running from a script from SCM, so manually list these for now.
+// Run the main test mix on all runs (PR + official)
+
+def innerLoopTests = [:]
+
+['x64', 'x86'].each { arch ->
+    [true,false].each { isBaseline ->
+        String baseline = ""
+        if (isBaseline) {
+            baseline = " baseline"
+        }
+        if (isPR() || !isBaseline) {
+            innerLoopTests["windows ${arch} ryujit full_opt pgo${baseline} perf"] = {
+               simpleNode('windows_server_2016_clr_perf', 180) {
```

Comment:
@mmitche do you know if I could move the calls to simpleNode to the functions above? I think it would make the code cleaner, but wasn't sure exactly how syntax here worked.